src/lib/cli_lib/background_daemon.ml

```diff
@@ -29,13 +29,25 @@ let run ~f port arg =
   in
   go Start
 
-let init ?(rest = false) ~f arg_flag =
+let rpc_init ~f arg_flag =
   let open Command.Param.Applicative_infix in
-  if rest then
-    Command.Param.return (fun rest_port arg () ->
-        let port = Option.value rest_port ~default:Port.default_rest in
-        f port arg )
-    <*> Flag.rest_port <*> arg_flag
-  else
-    Command.Param.return (fun port arg () -> run ~f port arg)
-    <*> Flag.port <*> arg_flag
+  Command.Param.return (fun port arg () -> run ~f port arg)
+  <*> Flag.port <*> arg_flag
+
+let graphql_init ~f arg_flag =
+  let open Command.Param.Applicative_infix in
+  Command.Param.return (fun rest_port arg () ->
+      let port = Option.value rest_port ~default:Port.default_rest in
+      let (module Graphql_client : Graphql_client_lib.S) =
+        ( module struct
+          include Graphql_client_lib.Make (struct
+            let address = "graphql"
+
+            let port = port
+
+            let headers = String.Map.empty
+          end)
+        end )
+      in
+      f (module Graphql_client : Graphql_client_lib.S) arg )
```

Comment:
Why pass the client here rather than accessing it globally?

Reply:
Client can't really be global because you need to pass in a port value to the functor now. So I thought it would be better to pass in this module rather than the port itself.

Reply:
But isn't the only reason you need to pass the port to the functor that you're not passing it to the function? I guess  it's not a dealbreaker but I don't really follow the argument.